Diagnostic/Utils/XmlUtil.py

```diff
@@ -33,6 +33,14 @@ def addElement(xml,path,el,selector=[]):
             continue
         element.append(el)
 
+
+def addElementToFirstMatchingPathOnly(xml, path, el, selector=[]):
+    element = xml.find(path)
+    if selector and element.get(selector[0]) != selector[1]:
+        return
+    element.append(el)
```

Comment:
If xml.find(path) didn't find anything, what happens when you try to do the append?